packages/gatsby-remark-images/src/gatsby-browser.js

```diff
@@ -0,0 +1,26 @@
+const { imageClass, imageBackgroundClass, imageWrapperClass } = require(`./classes`)
+
+exports.onRouteUpdate = () => {    
+    const imageWrappers = document.querySelectorAll(`.${imageWrapperClass}`)
+
+    Array.prototype.forEach.call(imageWrappers, (imageWrapper) => {
+        const backgroundElement = imageWrapper.querySelector(`.${imageBackgroundClass}`)
+        const imageElement = imageWrapper.querySelector(`.${imageClass}`)
+
+        // set the opacity to zero after we identify matching images
+        // to avoid issues where this script loads incorrectly
+        // resulting in hidden images
+        imageElement.style.opacity = 0
```

Comment:
I don't understand why you need to set opacity to 0 here

Reply:
the reason is to avoid breaking things when unexpected behaviour happens e.g. disabled JavaScript to make sure that the image is still visible in this case

Reply:
Won't this cause to be some weirdness with opacity transition?

Reply:
what about removing `transition` from html markup and adding it here? Also we should only play with `opacity` if image is not `complete` I think? Something like this:

```js
const onImageLoad = () => {
  imageElement.style.opacity = 0
  imageElement.style.transition = `opacity 0.5s`
  backgroundElement.style.opacity = 0
  imageElement.removeEventListener(`load`, onImageLoad)
}

if (image.complete) {
  backgroundElement.style.opacity = 0
} else {
  imageElement.style.opacity = 0
  imageElement.addEventListener(`load`, onImageLoad)
}
```

Reply:
Not that I noticed when I tested it but happy to remove if you're concerned

Reply:
I did test it and with `transition` typo below there was definitely some glitches :)

Reply:
sorry commented before your second comment. I'll have a tinker based on your suggestion :)